src/chocolatey/infrastructure.app/services/NugetService.cs

```diff
@@ -845,13 +845,58 @@ public void remove_rollback_directory_if_exists(string packageName)
             return packageInstalls;
         }
 
-        /// <summary>
-        /// Sets the configuration for the package upgrade
-        /// </summary>
-        /// <param name="config">The configuration.</param>
-        /// <param name="packageInfo">The package information.</param>
-        /// <returns>The original unmodified configuration, so it can be reset after upgrade</returns>
-        private ChocolateyConfiguration set_package_config_for_upgrade(ChocolateyConfiguration config, ChocolateyPackageInformation packageInfo)
+		public ConcurrentDictionary<string, PackageResult> get_outdated(ChocolateyConfiguration config)
+		{
+			var packageManager = NugetCommon.GetPackageManager(
+				config,
+				_nugetLogger,
+				_packageDownloader,
+				installSuccessAction: null,
+				uninstallSuccessAction: null,
+				addUninstallHandler: false);
+
+			var outdatedPackages = new ConcurrentDictionary<string, PackageResult>();
+
+			set_package_names_if_all_is_specified(config, () => { config.IgnoreDependencies = true; });
+			var packageNames = config.PackageNames.Split(new[] { ApplicationParameters.PackageNamesSeparator }, StringSplitOptions.RemoveEmptyEntries).or_empty_list_if_null().ToList();
+
+			foreach (var packageName in packageNames)
+			{
+				var latestPackage = packageManager.SourceRepository.GetPackages().Where(x => x.Id.ToLower() == packageName && x.IsLatestVersion).FirstOrDefault();

```

Comment:
I like it! How does it hold up against different sources though? Specifically I think this works great when it is hitting an OData feed like the community repository, but how does it work when it hits a file share or a NuGet.Server (Chocolatey.Server is a good one to test)?